src/pyhf/tensor/pytorch_backend.py

```diff
@@ -347,9 +348,16 @@ def poisson(self, n, lam):
         Returns:
             PyTorch FloatTensor: Value of the continuous approximation to Poisson(n|lam)
         """
-        return torch.exp(torch.distributions.Poisson(lam).log_prob(n))
+        # validate_args=True disallows continuous approximation
+        return torch.exp(
+            torch.distributions.Poisson(lam, validate_args=False).log_prob(n)
+        )
 
     def normal_logpdf(self, x, mu, sigma):
+        x = self.astensor(x)
+        mu = self.astensor(mu)
+        sigma = self.astensor(sigma)
```

Comment:
Do we want to do this? Or do we want to just switch up the public API to no longer accept floats, but only tensor objects?

Reply:
seems fine to do it this way. we will revisit the API later when we add type annotations.

Reply:
My only concern is that we did work to remove all the `self.astensor` calls that we could in PR #582. It might be worth this hit here to just get this fixed and a patch release out (as the canary tests are broken) and then force things to be tensors all around later.

Reply:
file an issue so we can get back to this later.

Reply:
Yup. I made #1352